### DISCUSSION TITLE
Reader Improvement

### DISCUSSION THREAD

**Discussion by Acclorite** [Acclorite/book-story#87] [Ideas]
Thread for [Reader Improvement milestone](https://github.com/Acclorite/book-story/milestone/1).
Feel free to share your ideas here.

**Comment by alcamiz** [comment]
Would you consider ditching custom parsing in favor of a different engine, [Readium](https://github.com/readium/kotlin-toolkit)? It's written in pure Kotlin, and should be able to handle most things you could find in epub3 ebooks. It already supports most improvements that were suggested in the milestone (images, pagination, text-to-speech, bookmarks), and additionally you can easily use it to support pdf files as well. You can also store books in a safe zone as you mentioned on #57, which you can test out with the test app they pre-build on their release page.

The entire [toolkit](https://readium.org) also has a ton of support, so development should continue to the foreseeable future. What do you think?

**Reply by Acclorite** [reply]
I am sorry for such a long response.. I have forgotten to answer this and so I am here. Creating custom parsers allows me to create more specific things and to not rely on something or someone else. Not to mention that such big change would be very big in terms of code.

**Reply by alcamiz** [reply]
No worries! Just thought I'd let you know in case you were interested. Thanks for responding!

**Comment by rvsmooth** [comment]
What would it be to get a inline word searching engine, as that of calibre or librera

**Reply by Acclorite** [reply]
Follow #104

**Reply by Acclorite** [reply]
If you talk about something like dictionary, then it is an online feature, the app is currently fully offline. (More info: #40)

**Reply by rvsmooth** [reply]
> If you talk about something like dictionary, then it is an online feature, the app is currently fully offline. (More info: #40)

I see. Fair reason.

**Comment by rabbitAiry** [comment]
I'm reading a book in foreign language recently, but I found that it's somehow a bit not convenient to get to the translation app since I have to choose the translate function in a hidden menu and I have to choose which app to translate every time. Anyway, thank you for making such a beautiful app and hope to get improved.

**Comment by yonesaku** [comment]
Let me start by saying i absolutely love the app! 

I was wondering if it is worth to consider having an option for costum fonts? Like choosing from a ttf file?

**Reply by Acclorite** [reply]
Hey, thank you! (sorry for the late response, didn't see)
There is an issue on that already (planned): #52

**Reply by yonesaku** [reply]
Oh! I didn't see that sorry for the duplicate. Looking forward to it!

**Comment by moonviellereader** [comment]
Hi, do u have Flipped mode (not scroll)? I search on the setting, but I can't find that option

**Reply by Acclorite** [reply]
Hey. Not yet. Follow: #13.
